src/providers/oracle/qgsoraclefeatureiterator.cpp

```diff
@@ -528,6 +528,29 @@ bool QgsOracleFeatureIterator::openQuery( const QString &whereClause, const QVar
   return true;
 }
 
+bool QgsOracleFeatureIterator::execQuery( const QString &query, const QVariantList &args, int retryCount )
+{
+  if ( !QgsOracleProvider::exec( mQry, query, args ) )
+  {
+    if ( retryCount != 0 )
+    {
+      // If the connection has been closed
+      // Try again N times in case of timeout
+      // ORA-12170: TNS:Connect timeout occurred
+      if ( mQry.lastError().number() == 12170 )
+      {
+        // restart connection
+        mConnection->reconnect();
+        // rebuild query with the new connection
+        mQry = QSqlQuery( *mConnection );
```

Comment:
is it necessary to recreate the query?

Reply:
You're right. It's not necessary to recreate the query.